### PR TITLE
ENH: Add a ``RobustAverage`` interface and split volume selection

### PR DESCRIPTION
**Pull request by oesteban**
Resolves: #600.

### CODE REVIEW

**Comment by oesteban** [comment]
I'll work on the interface that generates `t_list` tomorrow.

**Comment by oesteban** [comment]
Okay, this is looking better. Please LMKWYT, @mgxd .

File: niworkflows/interfaces/images.py [hunk 1]
```diff
@@ -188,6 +191,134 @@ def _run_interface(self, runtime):
...
+        t_mask = (
+            self.inputs.t_mask if isdefined(self.inputs.t_mask) else [True] * img_len
+        )
+
+        if len(t_mask) != img_len:
+            raise ValueError(
+                f"Image length ({img_len} timepoints) unmatched by mask ({len(t_mask)})"
+            )
+
+        n_volumes = np.sum(self.inputs.t_mask)
```
Comment (mgxd) [inline niworkflows/interfaces/images.py:262]: what if this is undefined?
```suggestion
        n_volumes = np.sum(t_mask)
```

File: niworkflows/interfaces/images.py [hunk 2]
```diff
@@ -188,6 +191,134 @@ def _run_interface(self, runtime):
...
+        if n_volumes < 1:
+            raise ValueError("At least one volume should be selected for slicing")
+
+        self._results["out_file"] = fname(suffix="_average")
+        self._results["out_volumes"] = fname(suffix="_sliced")
+
+        sliced = nb.concat_images(
+            i for i, t in zip(nb.four_to_three(img), self.inputs.t_mask) if t
```
Comment (mgxd) [inline niworkflows/interfaces/images.py:270]:
```suggestion
            i for i, t in zip(nb.four_to_three(img), t_mask) if t
```

**Review by mgxd — APPROVED** [review]
Few comments, but this LGTM

File: niworkflows/interfaces/images.py (hunk 1 above)
Comment (oesteban) [reply niworkflows/interfaces/images.py:262]: Yup, this was the idea - thanks for catching it.

**Pull request merged**